src/transformLoopWithLimit.js

```diff
@@ -1,26 +1,38 @@
 'use strict'
 
 module.exports = function transformLoopWithLimit (limit) {
-  return function transformLoop (node) {
-    const loopKeywords = ['WhileStatement', 'ForStatement', 'DoWhileStatement']
+  const loopNodeTypes = ['WhileStatement', 'ForStatement', 'DoWhileStatement']
+
+  const isUnlabeledLoop = (node) =>
+    loopNodeTypes.indexOf(node.type) > -1 && node.parent.type !== 'LabeledStatement'
+
+  const isLabeledLoop = (node) =>
+    loopNodeTypes.indexOf(node.type) > -1 && node.parent.type === 'LabeledStatement'
+
+  const isBodyOfLoop = (node) =>
+     node.type === 'BlockStatement' && loopNodeTypes.indexOf(node.parent.type) > -1
 
-    if (loopKeywords.indexOf(node.type) > -1) {
-      node.update(
-        'var __ITER = ' + limit + ';' +
-        node.source()
-      )
+  const addVarDeclarationBeforeNode = (node) =>
+    node.update('var __ITER = ' + limit + ';' + node.source())
```

Comment:
As we are refactoring this, what are your thoughts on using es6 string templates?